celery/canvas.py

```diff
@@ -2248,13 +2247,14 @@ def link_error(self, errback):
             applied to the body.
         """
         if self.app.conf.task_allow_error_cb_on_chord_header:
-            # self.tasks can be a list of the chord header workflow.
-            if isinstance(self.tasks, (list, tuple)):
-                for task in self.tasks:
-                    task.link_error(errback)
-            else:
-                self.tasks.link_error(errback)
+            for task in self.tasks:
+                task.link_error(errback)
```

Comment:
Why did you remove this condition? `self.tasks` may be a list or a tuple and if it isn't, it's a group.

E.g.
```
chord([s1.s(), s2.s()], body.s())
chord(group(s1.s(), s2.s()), body.s())
```

We need to support both cases.

Reply:
There's no problem with that, `self.tasks` is accessed correctly with both examples.
To prove it, I've added the PR a test case with your both examples: `test_link_error_on_chord_header()`
Check it out to confirm and resolve conversation if you're satisfied @thedrow